geoh5py/data/referenced_data.py

```diff
@@ -39,6 +43,42 @@ def __init__(self, **kwargs):
 
         super().__init__(**kwargs)
 
+    def copy(
+        self,
+        parent=None,
+        *,
+        clear_cache: bool = False,
+        mask: np.ndarray | None = None,
+        **kwargs,
+    ) -> Data:
+        """
+        Overwrite the copy method to ensure that the uid is set to None.
+        """
+        kwargs["uid"] = None
+
+        new_data = cast(
+            ReferencedData,
+            super().copy(
+                parent=parent,
+                clear_cache=clear_cache,
+                mask=mask,
+                omit_list=["_metadata", "_data_maps"],
+                **kwargs,
+            ),
+        )
+
+        # Always overwrite the entity type name to protect the GeometricDataValueMapType
+        new_data.entity_type.name = find_unique_name(
+            self.entity_type.name,
+            [tp.name for tp in self.workspace.types if isinstance(tp, DataType)],
+        )
+
+        if self.data_maps is not None:
+            for name, child in self.data_maps.items():
```

Comment:
data_maps has a setter. You could pass directly 
`new_data.data_maps=self.data_maps` no?

Reply:
The data map setter accepts a dictionary of GeometricData. There are more mechanics involved with the `add_data_map`, such as creating the GeometricData and its data type, which needs to be linked to the reference data. The linkage between the ReferenceData and the GeometricData is pretty complicated on geoh5, and I raised the issue to @krisdavis. We can simplify it once we have a stronger link.

Reply:
ok, maybe you could create an issue then? 
And precise to change the loops when it's ready?